modules/issuer/main.tf

```diff
@@ -0,0 +1,125 @@
+# create certificates for the trust anchor and issuer
+resource "tls_private_key" "this" {
+  for_each    = local.certs
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P256"
+}
+
+# Control Plane TLS Credentials
+resource "tls_self_signed_cert" "this" {
+  for_each = local.certs
+
+  key_algorithm     = tls_private_key.this[each.key].algorithm
+  private_key_pem   = tls_private_key.this[each.key].private_key_pem
+  is_ca_certificate = true
+
+  validity_period_hours = var.trust_anchor_validity_hours
+
+  allowed_uses = each.value.allowed_uses
+
+  subject {
+    common_name = "${each.key}.linkerd.cluster.local"
+  }
+}
+
+
+# create namespaces for linkerd and any extensions (linkerd-viz or linkerd-jaeger)
+resource "kubernetes_namespace" "namespace" {
+  for_each = toset(local.namespaces)
+  metadata {
+    name        = each.key
+    annotations = (each.key != var.namespace) ? { "linkerd.io/inject" = "enabled" } : {}
```

Comment:
I know this is legacy, but not sure why we are setting the "inject" annotation on the linkerd namespaces themselves.

Reply:
@appkins - Some of the checks are failing.  Thoughts?

Reply:
@appkins Did you test stopping and restarting the cluster with HA installed?

Reply:
When running "linkerd check" I see an issue with the certs:
` 1m×× tap API server has valid cert
    cert is not issued by the trust anchor: x509: certificate is valid for tap.linkerd.svc, not tap.linkerd-viz.svc
    see https://linkerd.io/checks/#l5d-tap-cert-valid for hints`

Reply:
I made a bunch of changes last week @ggolub 
I was seeing the same issues, but now am able to deploy HA linkerd that works with vault injector. The latest changes move to the edge release of linkerd that provides two helm charts for the deployment. Linkerd now has a crd deployment and a core component. This allows us to deploy linkerd crds with no dependencies, create custom resources, then deploy the meat of linkerd. The deployments seem to be much more stable after the last batch of changes.

Reply:
Another thing worth mentioning, linkerd inject annotations are just an effect of the namespace selectors provided in the helm chart. I'm now overriding the namespace selector for proxyinjector which was causing issues before on linkerd namespaces.

```terraform
proxyInjector = {
      namespaceSelector = {
        matchExpressions = [
          { key = "linkerd.io/is-control-plane", operator = "DoesNotExist" },
          { key = "linkerd", operator = "In", values = ["enabled"] }
        ]
      }
    }
```

This means that a resource (namespace) with the label `linkerd: enabled` will be processed by proxy injector. We have a lot of flexibility and might want to review how we set up the selectors before merging this PR.

Reply:
> I know this is legacy, but not sure why we are setting the "inject" annotation on the linkerd namespaces themselves. I think this was a day-one error that should be removed. We don't need lines 31 and 32

Good idea, we could actually add the annotation `linkerd.io/is-control-plane` and explicitly disable proxy injector for linkerd namespaces, allowing linkerd to be enabled by default everywhere else in the cluster. How do you feel about that solution for a default path?